### PR TITLE
Simplify Iterator APIs

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/SmartContract/FindOptions.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Neo.SmartContract
+{
+    [Flags]
+    public enum FindOptions : byte
+    {
+        None = 0,
+
+        KeysOnly = 1 << 0,
+        RemovePrefix = 1 << 1,
```
Comment (shargon) [inline src/neo/SmartContract/FindOptions.cs:11]: `RemovePrefix` before `KeysOnly`?

File: src/neo/SmartContract/Iterators/StorageIterator.cs [hunk 2]
```diff
@@ -7,30 +8,48 @@ namespace Neo.SmartContract.Iterators
...
+            byte[] value = enumerator.Current.Value.Value;
+
+            if (options.HasFlag(FindOptions.RemovePrefix))
+                key = key[1..];
+
+            StackItem item = options.HasFlag(FindOptions.DeserializeValues)
+                ? BinarySerializer.Deserialize(value, 1024, (uint)value.Length, referenceCounter)
+                : value;
+
+            if (options.HasFlag(FindOptions.PickField0))
```
Comment (shargon) [inline src/neo/SmartContract/Iterators/StorageIterator.cs:43]: If has both flags? `PickField0` and `PickField1`?

Comment (erikzhang) [reply src/neo/SmartContract/Iterators/StorageIterator.cs:43]: https://github.com/neo-project/neo/blob/b29e24e5b64db399aca52ca6172407846bb5e25c/src/neo/SmartContract/ApplicationEngine.Storage.cs#L67-L68

**Comment by erikzhang** [comment]
Go?

**Review by shargon — APPROVED** [review]

**Pull request merged**